compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -351,7 +351,23 @@ object Trees {
       if (span.exists) {
         val point = span.point
         if (rawMods.is(Synthetic) || name.toTermName == nme.ERROR) Span(point)
-        else Span(point, point + name.stripModuleClassSuffix.lastPart.length, point)
+        else {
+          val realName = name.stripModuleClassSuffix.lastPart.toString
+          val nameStart =
+            if (point != span.start) point
+            else {
+              // Point might be too far away from start to be recorded. In this case we fall back to scanning
```

Comment:
Do we have a test case that exercises this ?

Reply:
The original test case was ErrorMessagesID, and I verified that the change here works for it. But c8265f1 fixes the problem by another means, so  it's no longer a test case.